utils/DeployUtils.ps1

```diff
@@ -0,0 +1,240 @@
+        [Parameter(Mandatory=$false)]
+        [AllowEmptyString()]
+        [string]
+        $OverrideModuleVersion = ""
+    )
+    #TODO: Add any module configuration needed (e.g., adjust Module Version)
+
+    $ManifestPath = Join-Path -Path $ModulePath -ChildPath "ScubaGear.psd1"
+    $ModuleVersion = $OverrideModuleVersion
+
+    if ([string]::IsNullOrEmpty($OverrideModuleVersion)){
+        $CurrentModuleVersion = (Import-PowerShellDataFile $ManifestPath).ModuleVersion
+        $TimeStamp = [int32](Get-Date -UFormat %s)
+        $ModuleVersion = "$CurrentModuleVersion.$TimeStamp"
+    }
+
+    $ManifestUpdates = @{
+        Path = $ManifestPath
+        ModuleVersion = $ModuleVersion
+        ProjectUri = "https://github.com/cisagov/ScubaGear"
+        LicenseUri = "https://github.com/cisagov/ScubaGear/blob/main/LICENSE"
+        Tags = 'CISA', 'AzureAD', 'Cloud', 'Configuration', 'Exchange', 'Report', 'Security', 'SharePoint', 'Defender', 'Teams', 'PowerPlatform'
+    }
+
+    Update-ModuleManifest @ManifestUpdates
+
+    $CurrentErrorActionPreference = $ErrorActionPreference
+    $ErrorActionPreference = "SilentlyContinue"
+    $Result = Test-ModuleManifest -Path $ManifestPath
+    $ErrorActionPreference = $CurrentErrorActionPreference
+
+    return $null -ne $Result
+}
+
+function SignScubaGearModule{
+    <#
+        .NOTES
+        Internal helper function
+    #>
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path -Path $_})]
+        $ModulePath,
+        [Parameter(Mandatory=$false)]
+        [scriptblock]
+        $GetCertificate = $function:GetCodeSigningCertificate,
+        [Parameter(Mandatory=$false)]
+        [ValidateScript({[uri]::IsWellFormedUriString($_, 'Absolute') -and ([uri] $_).Scheme -in 'http','https'})]
+        $TimeStampServer = 'http://timestamp.digicert.com',
+        [Parameter(Mandatory=$false)]
+        [ValidateSet('SHA256')]
+        [string]
+        $HashAlgorithm = 'SHA256'
+    )
+    #TODO: Add code signing
+    $CatalogFileName = 'ScubaGear.cat'
+    $CatalogPath = Join-Path -Path $ModulePath -ChildPath $CatalogFileName
+    $Cert = Invoke-Command $GetCertificate
+
+    # Digitally sign scripts, manifest, and modules
+    Get-ChildItem $ModulePath -Include *.psd1,*psm1,*.ps1 -Recurse |
+    Set-AuthenticodeSignature -Certificate $Cert -TimeStampServer $TimeStampServer -HashAlgorithm $HashAlgorithm
+
+    New-FileCatalog -Path $ModulePath -CatalogFilePath $CatalogPath -CatalogVersion 2.0 -Verbose
+    Set-AuthenticodeSignature -FilePath $CatalogPath -Certificate $Cert -TimestampServer $TimeStampServer -HashAlgorithm $HashAlgorithm
+
+    $TestResult = Test-FileCatalog -CatalogFilePath $CatalogPath
+    return 'Valid' -eq $TestResult.Status
+}
+
+function New-CodeSigningCertificate{
+    <#
+        .NOTES
+        Internal helper function
+    #>
+    New-SelfSignedCertificate -DnsName cisa.gov -Type CodeSigning -CertStoreLocation Cert:\CurrentUser\My
```

Comment:
This cmdlet publishes the thumbprint (and other info) to the logs.  Addam recently requested on one my PRs that I avoid publishing thumbprints to the logs.  He said, "While not particularly sensitive on its own, ideally this info would not be printed. Suggest redirecting output to null since we don't need the output in the log..."  The solution that he recommended to me was to add "| Out-Null" to the end of the command, which worked for me.